src/app/cluster/details/external-cluster/update-external-cluster-machine-deployment-dialog/component.ts

```diff
@@ -0,0 +1,134 @@
+}
+
+@Component({
+  selector: 'km-update-external-cluster-machine-deployment-cluster-dialog',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class UpdateExternalClusterMachineDeploymentDialogComponent extends BaseFormValidator implements OnInit {
+  readonly Controls = Controls;
+  disableReplicaControl: boolean;
+  versions: string[] = [];
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: UpdateExternalClusterMachineDeploymentDialogData,
+    private readonly _dialogRef: MatDialogRef<UpdateExternalClusterMachineDeploymentDialogComponent>,
+    private readonly _notificationService: NotificationService,
+    private readonly _externalMachineDeploymentService: ExternalMachineDeploymentService,
+    private readonly _builder: FormBuilder
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    this._initForm();
+    this._initSubscriptions();
+  }
+
+  getObservable(): Observable<any> {
+    const patchModel = this._updateExternalMachineDeploymentPatchModel();
+    const {projectID, clusterID, machineDeployment} = this.data;
+    return this._externalMachineDeploymentService
+      .patchExternalMachineDeployment(projectID, clusterID, machineDeployment.id, patchModel)
+      .pipe(take(1));
+  }
+
+  onNext(md: ExternalMachineDeployment): void {
+    this._notificationService.success(`Updated the ${md.name} machine deployment`);
+    this._dialogRef.close();
+  }
+
+  private _initForm(): void {
+    this.form = this._builder.group({
+      [Controls.Replicas]: this._builder.control(1),
+      [Controls.KubeletVersion]: this._builder.control(''),
+    });
+  }
+
+  private _initSubscriptions(): void {
+    this.form.valueChanges.pipe(takeUntil(this._unsubscribe)).subscribe(_ => {
+      this._updateExternalMachineDeploymentPatchModel();
+    });
+
+    const initialValue = this.form.value;
+    this.form
+      .get(Controls.Replicas)
+      .valueChanges.pipe(takeUntil(this._unsubscribe))
+      .subscribe(value => {
+        if (initialValue[Controls.Replicas] === value) {
+          this.form.get(Controls.KubeletVersion).enable({emitEvent: false});
+        } else {
+          this.form.get(Controls.KubeletVersion).disable({emitEvent: false});
+        }
+      });
+
+    this.form
+      .get(Controls.KubeletVersion)
+      .valueChanges.pipe(takeUntil(this._unsubscribe))
+      .subscribe(value => {
+        if (initialValue[Controls.KubeletVersion] === value) {
+          this.disableReplicaControl = false;
+        } else {
+          this.disableReplicaControl = true;
+        }
+      });
+
+    this._externalMachineDeploymentService
+      .machineDeploymentUpgrades(this.data.projectID, this.data.clusterID, this.data.machineDeployment?.id)
+      .pipe(take(1))
+      .subscribe((upgrades: MasterVersion[]) => {
+        this.versions = upgrades.map(upgrade => upgrade.version);
+      });
+  }
+
+  private _updateExternalMachineDeploymentPatchModel(): ExternalMachineDeploymentPatch {
```

Comment:
```suggestion
  private _getExternalMachineDeploymentPatchModel(): ExternalMachineDeploymentPatch {
```
Since this method is returning a value.